compiler/rustc_ast_lowering/src/lib.rs

```diff
@@ -140,6 +148,12 @@ struct LoweringContext<'a, 'hir> {
     generics_def_id_map: Vec<FxHashMap<LocalDefId, LocalDefId>>,
 }
 
+struct LoweringArena {
```

Comment:
Could you use `rustc_arena::declare_arena`?

Reply:
Done

Reply:
I meant using the macro `rustc_arena::declare_arena`.

Reply:
I've tried to use `declare_arena` but that imposes a `'tcx` lifetime which any of the fields have. At that point I've tried `_marker: PhantomData<&'tcx ()>` but then I got a bunch of lifetime errors and I ended thinking that using the macro may be more complicated than writing this manually.
Let me know if you still want to persue this or this is ready to be merged in the way it is.